# patch target libc, returns *-unknown-linux-gnu on musl builds without --static
--- a/vendor/github.com/influxdata/pkg-config/libs/flux/build.go	2026-09-07 10:06:29.318714673 -0400
+++ b/vendor/github.com/influxdata/pkg-config/libs/flux/build.go	2026-09-07 10:06:29.319620871 -0400
@@ -43,24 +43,27 @@
 
 // Determine the cargo target.
 func (t Target) DetermineCargoTarget(logger *zap.Logger) string {
+	musl := t.Static || os.Getenv("XBPS_TARGET_LIBC") == "musl"
 	switch {
-	case t.OS == "linux" && t.Arch == "amd64" && t.Static:
+	case t.OS == "linux" && t.Arch == "amd64" && musl:
 		return "x86_64-unknown-linux-musl"
-	case t.OS == "linux" && t.Arch == "amd64" && !t.Static:
+	case t.OS == "linux" && t.Arch == "amd64" && !musl:
 		return "x86_64-unknown-linux-gnu"
-	case t.OS == "linux" && t.Arch == "386" && !t.Static:
+	case t.OS == "linux" && t.Arch == "386" && musl:
+		return "i686-unknown-linux-musl"
+	case t.OS == "linux" && t.Arch == "386" && !musl:
 		return "i686-unknown-linux-gnu"
-	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "6" && !t.Static:
+	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "6" && !musl:
 		return "arm-unknown-linux-gnueabihf"
-	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "6" && t.Static:
+	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "6" && musl:
 		return "arm-unknown-linux-musleabihf"
-	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "7" && !t.Static:
+	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "7" && !musl:
 		return "armv7-unknown-linux-gnueabihf"
-	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "7" && t.Static:
+	case t.OS == "linux" && t.Arch == "arm" && t.Arm == "7" && musl:
 		return "armv7-unknown-linux-musleabihf"
-	case t.OS == "linux" && t.Arch == "arm64" && !t.Static:
+	case t.OS == "linux" && t.Arch == "arm64" && !musl:
 		return "aarch64-unknown-linux-gnu"
-	case t.OS == "linux" && t.Arch == "arm64" && t.Static:
+	case t.OS == "linux" && t.Arch == "arm64" && musl:
 		return "aarch64-unknown-linux-musl"
 	case t.OS == "linux" && t.Arch == "s390x":
 		return "s390x-unknown-linux-gnu"
